cmd/libsnap-confine-private/panic.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "panic.h"
+
+#include <errno.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+static sc_panic_exit_fn panic_exit_fn = NULL;
+static sc_panic_msg_fn panic_msg_fn = NULL;
+
+void sc_panic(const char *fmt, ...) {
```

Comment:
Quick drive-by: maybe some else than "panic"? Panic smells a lot like a go panic() which will behave differently AFAICT (print stacktrace etc).

Reply:
Well, the fact that it prints a stack trace isn't the essential feature. We could also print one if that would make you feel better. I'm somewhat -1 on this precisely because of error + panic parity that C and Go can have.

Reply:
I'm not arguing in favour of adding a stacktrace to the C code. Just that its confusing to use the same name for something different without any advantage in doing so (at least I don't see one).

Reply:
I argue that the principal function of `sc_panic` is the same as `panic`, to abort execution without retaining regular control flow.